pkg/arvo/sys/vane/ames.hoon

```diff
@@ -3668,13 +3767,500 @@
+                ::
+                ++  veri
+                  |=  [sig=@ dat=@]
+                  ^-  ?
+                  (safe:as:(com:nu:crub:crypto public-key.peer-state) sig dat)
+                ::
+                ++  meri
+                  |=  [pax=path sig=@ dat=$@(~ (cask))]
+                  (veri sig (mess her life.peer-state pax dat))
+                --
+              --
+          ::
+          |_  [=path keen=keen-state]
+          ::
+          +|  %helpers
+          ::
+          ++  fine  .
+          ++  abed
+            |=  p=^path
+            ~|  no-keen-for-path/p
+            fine(path p, keen (~(got by keens) p))
+          ::
+          ++  fi-abet
+            ^+  peer-core
+            ?.  =,  keen
+                ::  num-fragments is 0 when unknown (i.e. no response yet)
+                ::  if no-one is listening, kill request
+                ::
+                ?|  =(~ listeners.keen)
+                    &(!=(0 num-fragments) =(num-fragments num-received))
+                ==
+              =.  fine   fi-set-wake
+              peer-core(keens.peer-state (~(put by keens) path keen))  :: XX tack.keens
+            ::
+            =?  fine  ?=(^ next-wake.keen)
+              (fi-rest u.next-wake.keen)
+            peer-core(keens.peer-state (~(del by keens) path))  :: XX tack.keens
+          ::
+          ++  fi-full-path
+            :^    (scot %p her)
+                (scot %ud rift.peer-state)
+              (scot %ud life.peer-state)
+            path
+          ::
+          ++  fi-show
+            =,  keen
+            :*  nex=(lent nex)
+                hav=(lent hav)
+                num-fragments=num-fragments
+                num-received=num-received
+                next-wake=next-wake
+                metrics=metrics
+            ==
+          ::
+          ++  fi-trace
+            |=  [verb=? print=(trap tape)]
+            ^+  same
+            (trace %fine verb her ships.bug.ames-state print)
+          ::
+          ++  fi-emit       |=(move fine(event-core (emit +<)))
+          ++  fi-deq        (deq want)
+          ++  fi-gauge      (ga metrics.keen (wyt:fi-deq wan.keen))
+          ++  fi-wait       |=(tim=@da (fi-pass-timer %b %wait tim))
+          ++  fi-rest       |=(tim=@da (fi-pass-timer %b %rest tim))
+          ++  fi-etch-keen  |=(frag=@ud (etch-keen fi-full-path frag))
+          ++  fi-send
+            |=(=hoot fine(event-core (send-blob for=| her `@ux`hoot)))
+          ::
+          ++  fi-clean-up
+            |=  [=^duct =_fine]
+            ::  at the moment only %spider keeps state about remote scries
+            ::
+            ?+  duct  fine
+                [[%gall %use %spider @ ship=@ %thread tid=@ *] *]
+              =/  =cage   spider-stop+!>([&7.-.duct |])
+              =/  poke=*  [%0 %m [p q.q]:cage]
+              =/  =plea   [%g /ge/spider poke]
+              (fi-emit:fine duct %pass /fine/unsub %g %plea our plea)
+            ==
```

Comment:
This pattern is not sustainable. Other clients need to be able to add/remove such state without upfront coordination through %ames. A %miss should be given on all these ducts.

Reply:
Pushed this https://github.com/urbit/urbit/pull/6406/commits/873de612691baba42dbce62349488fa7af4fb6b6 addressing your request, let me know how it looks.